lib/commands/eval.js

```diff
@@ -0,0 +1,31 @@
+module.exports = {
+    permissions: ["developer"],
+    guildOnly: true,
+    cooldown: 5000,
+    shortDescription: 'Evaluates a command',
+    longDescription: 'Evaluates arbitrary code. Only developers have access to this command.',
+    run: async function(client, m, args) {
+        let expression = args.join(" ");
+        try {
+            let result = eval(expression);
+
+            // Result returned a Promise
+            if(result.toString() === "[object Promise]") {
+                let now = new Date().getTime();
+                result.then((...values) => {
+                    let duration = (new Date().getTime()) - now;
+                    m.channel.createMessage(`\`\`\`js\nIn:\n${expression}\nOut:\nPromise resolved after ${duration}ms\nOut:${values}\n\`\`\``);
+                }).catch((err) => {
+                    let duration = (new Date().getTime()) - now;
```

Comment:
Just a note, but why not use `const` by default and then change to `let` if you absolutely need to modify the value?

This could prevent some mistakes where you'd accidentally modify a value when it wasn't needed.

Reply:
Habit more than anything really. I tend to use `const` if the variable it's attached to will always have the same value.